docs/paper/dev/api/mob-goals.md

```diff
@@ -0,0 +1,126 @@
+---
+slug: /dev/mob-goals
+description: A guide to the mob goal API.
+---
+
+# Mob Goal API
+
+The Mob Goal API is a way of controlling the behavior of mobs in Minecraft. It allows you to set a goal for a mob to perform, such as
+attacking a player, or moving to a location. It also allows you to create your own custom goals.
+
+## Registering a Goal on an Entity
+
+To register a goal on an entity, you need to create an instance of the goal and then register it with the entity:
+
+```java
+Cow cow = ...;
+Goal<Cow> goal = new ExampleGoal();
+
+server.getMobGoals().addGoal(cow, 0, goal); // 0 is the priority, lower numbers are higher priority
+```
+
+:::tip
+
+You can access the Vanilla goals from the `VanillaGoal` class. These are the goals that are used by Vanilla Minecraft.
+They are specific to each mob type, so you can't use `VanillaGoal.BEE_ATTACK` on a Zombie, for example.
+
+:::
```

Comment:
This feels confusing as it kinda reads as if one could add Vanilla goals to entities (especially under the `addGoal` example) which isn't possible right now.